pyomo/contrib/doe/doe.py

```diff
@@ -826,7 +928,9 @@ def initialize_jac(m, i, j):
             # Otherwise initialize to 0.1 (which is an arbitrary non-zero value)
             else:
                 raise AttributeError(
```

Comment:
You may want to have this be a `DeveloperError` or `MouseTrap` error instead of an `AttributeError`.